drivers/display/ls0xx.c

```diff
@@ -0,0 +1,441 @@
+
+	return gpio_pin_set(driver->disp_dev, DT_INST_GPIO_PIN(0, disp_gpios), 0);
+#else
+	LOG_WRN("Unsupported");
+	return -ENOTSUP;
+#endif
+}
+
+/**
+ * @brief Clear the internal memory of the display
+ *
+ * @details Clear the display memory and the frame_buf
+ *          setting it to all white
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_clear(const struct device *dev)
+{
+	struct ls0xx_data *driver = dev->data;
+	int err;
+	uint8_t clear_cmd[2] = { LS0XX_BIT_CLEAR | LS0XX_EXTCOMIN_BIT, 0 };
+
+	tx_buf.len = sizeof(uint16_t);
+	tx_buf.buf = &clear_cmd;
+	err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+	spi_release(driver->spi_dev, &driver->spi_config);
+
+	return err;
+}
+
+/**
+ * @brief Send required lines to the display
+ *
+ * @details Update the display by sending num_lines
+ *          starting from start_line.
+ *
+ * @param dev Pointer to the display device
+ * @param start_line Starting line number, starts with 1
+ *                   corresponds to Y co-ordinate + 1
+ * @param num_lines Number of lines ie height to update
+ * @param buf  Buffer containing data to write
+ */
+static int ls0xx_update_display(const struct device *dev,
+				uint16_t start_line,
+				uint16_t num_lines,
+				const uint8_t *buf)
+{
+	struct ls0xx_data *driver = dev->data;
+
+	int err;
+	uint8_t write_cmd;
+	uint16_t max_line;
+
+	LOG_DBG("Lines %d to %d", start_line, start_line + num_lines - 1);
+
+	write_cmd = LS0XX_BIT_WRITECMD | LS0XX_EXTCOMIN_BIT;
+#if defined(CONFIG_LS0XX_VCOM_DRIVER) && !DT_INST_NODE_HAS_PROP(0, extcomin_gpios)
+	extcomin_bit = LS0XX_EXTCOMIN_BIT ? 0x00 : LS0XX_BIT_VCOM;
+#endif
+	tx_buf.len = sizeof(uint8_t);
+	tx_buf.buf =  &write_cmd;
+	err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+	max_line = start_line + num_lines - 1;
+
+	if (buf != NULL) {
+		/* Send each line to the screen including
+		 * the line number and dummy bits
+		 */
+		tx_buf.len = sizeof(uint8_t) * LS0XX_BYTES_PER_LINE;
+		tx_buf.buf = &line_buf[0];
+		for (int l = start_line; l <= max_line; l++) {
+			line_buf[0] = l;
+			memcpy(&line_buf[1], buf, LS0XX_PANEL_WIDTH / LS0XX_PIXELS_PER_BYTE);
+			buf += LS0XX_PANEL_WIDTH / LS0XX_PIXELS_PER_BYTE;
+			err |= spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+		}
+
+		/* Send another trailing 8 bits for the last line
+		 * These can be any bits, it does not matter
+		 * just reusing the write_cmd buffer
+		 */
+		tx_buf.len = sizeof(uint8_t);
+		tx_buf.buf =  &write_cmd;
+		err |= spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+	}
+	spi_release(driver->spi_dev, &driver->spi_config);
+
+	return err;
+}
+
+/**
+ * @brief Write to the display at given co-ordinates
+ *
+ * @details Write buffer to the display.
+ *          Buffer width should be a multiple of 8
+ *
+ *          If CONFIG_LS0XX_VCOM_DRIVER=y and extcomin-gpios is not defined
+ *          and buffer is set to NULL, or height/width is 0 then
```

Comment:
Comment is not correct  if width would be set to zero an error is returned, see line 240